code/controllers/subsystem/aura.dm

```diff
@@ -5,11 +5,48 @@ SUBSYSTEM_DEF(aura)
 	flags = SS_NO_INIT
 	///Stores currently active aura_bearer datums
 	var/list/active_auras = list() //We can't use a normal subsystem processing list because as soon as an aura_bearer leaves the list it needs to die
+	///Stores what the current aura processing stage was working on for if it is paused.
+	var/list/current_cache = list()
+	///Auras go through three stages. Pulse auras -> finish xeno cycles -> finish human cycles. Depending on where it was paused, this tells it where to resume fire()
+	var/stage = 1
 
 /datum/controller/subsystem/aura/fire(resumed)
-	finish_aura_cycles()
-	for(var/datum/aura_bearer/bearer AS in active_auras)
-		bearer.process()
+	if(stage == 1)
+		if(!resumed)
+			current_cache = active_auras.Copy()
+		while(current_cache.len)
+			var/datum/aura_bearer/bearer = current_cache[current_cache.len]
+			current_cache.len--
+			if(QDELETED(bearer))
+				continue
+			bearer.process()
+			if(MC_TICK_CHECK)
+				return
+		stage = 2
+	if(stage == 2)
+		if(!resumed)
+			current_cache = GLOB.xeno_mob_list.Copy()
```

Comment:
If you resume during stage one this'll never happen.

Reply:
oh, yeah makes sense.

Reply:
Should be resolved.